includes/Checker/Checks/Abstract_PHP_CodeSniffer_Check.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks\PHP_CodeSniffer_Check
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker\Checks;
+
+use WordPress\Plugin_Check\Checker\Check;
+use WordPress\Plugin_Check\Checker\Check_Result;
+use Exception;
+
+/**
+ * Check for running one or more PHP CodeSniffer sniffs.
+ *
+ * @since 1.0.0
+ */
+abstract class Abstract_PHP_CodeSniffer_Check implements Check {
+
+	/**
+	 * List of allowed PHPCS arguments.
+	 *
+	 * @var array
+	 */
+	protected $allowed_args = array(
+		'extensions',
+		'standard',
+		'sniffs',
+		'exclude',
+	);
+
+	/**
+	 * Returns an associative array of arguements to pass to PHPCS.
+	 *
+	 * @return array
+	 */
+	abstract public function get_args();
+
+	/**
+	 * Creates the command arguments.
+	 *
+	 * @param Check_Result $result The check results including the plugin context to check.
+	 *
+	 * @return array
+	 */
+	private function create_args( Check_Result $result ) {
+		// Set the default arguments for PHPCS.
+		$argv = array(
+			'',
+			$result->plugin()->path( '' ),
+			'--report=Json',
+			'--report-width=9999',
+		);
+
+		// Only accept allowed PHPCS arguments from check arguments array.
+		$check_args = array_filter(
+			$this->get_args(),
+			function( $key ) {
+				return in_array( $key, $this->allowed_args, true );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
+
+		// Format check arguments for PHPCS.
+		foreach ( $check_args as $key => $value ) {
+			$argv[] = "--{$key}=$value";
+		}
+
+		return $argv;
+	}
+
+	/**
+	 * Amends the given result by running the check on the associated plugin.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 *
+	 * @throws Exception Thrown when the check fails with a critical error (unrelated to any errors detected as part of
+	 *                   the check).
+	 */
+	public function run( Check_Result $result ) {
+		// Include the PHPCS autoloader.
+		$autoloader = TESTS_PLUGIN_DIR . '/vendor/squizlabs/php_codesniffer/autoload.php';
```

Comment:
Should `WP_PLUGIN_CHECK_PLUGIN_DIR_PATH` instead of `TESTS_PLUGIN_DIR`

```suggestion
		$autoloader = WP_PLUGIN_CHECK_PLUGIN_DIR_PATH . '/vendor/squizlabs/php_codesniffer/autoload.php';
```